dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -277,7 +277,7 @@ pub contract DSSCollection: NonFungibleToken {
 
         // Mint a DSSCollection NFT in this group
         //
-        pub fun mint(completionAddress: String, level: UInt8): @DSSCollection.NFT {
+        pub fun mint(completionAddress: String, level: UInt8, recipientAddress: Address): @DSSCollection.NFT {
```

Comment:
isn't the completion address the same as the recipient address?

Reply:
`completionAddress` is a **String** type and `recipientAddress` is an **Address** type... i don't think you can convert a String -> Address in cadence, but let me re-confirm i may be wrong

Reply:
@judezhu I tried to leverage `AddressUtils` https://flowscan.org/contract/A.a340dc0a4ec828ab.AddressUtils but hitting computation exceeds limit (100) ... it's not ideal since we have to parse the address everytime before calling `getCompletedCollectionIDs`.

maybe having identical addresses here for our use case is better for performance 🤔

Reply:
Is it possible that we just import the code here:

```
    pub fun parseAddress(_ input: AnyStruct): Address?{
        if let parsed = self.parseUInt64(input){
            return Address(parsed)
        }
        return nil
    }
```
this change will break the local e2e test, since it's adding a dependency of a new contract. cc @stevenhertel

Reply:
sadly no :( that `self.parseUInt64`  call requires the full dependency tree (`StringUtils`, `ArrayUtils` and `AddressUtils`)

We could go back to my initial approach of just having two version of address in the mint call, an `Address` and an `String`. The only downside there is we have to update the mint/mint_record transactions to include that extra Address variable?